packages/server-wallet/e2e-test/payer/client.ts

```diff
@@ -12,17 +12,26 @@ import {defaultConfig, ServerWalletConfig} from '../../src/config';
 import {ONE_DAY} from '../../src/__test__/test-helpers';
 
 export default class PayerClient {
-  private readonly wallet: ServerWallet;
-  constructor(
+  readonly config: ServerWalletConfig;
+  private constructor(
     private readonly pk: Bytes32,
     private readonly receiverHttpServerURL: string,
-    readonly config?: ServerWalletConfig
+    private readonly wallet: ServerWallet
   ) {
-    this.wallet = recordFunctionMetrics(
-      ServerWallet.create(this.config || payerConfig),
+    this.config = wallet.walletConfig;
+  }
+  public static async create(
+    pk: Bytes32,
+    receiverHttpServerURL: string,
+    config?: ServerWalletConfig
+  ): Promise<PayerClient> {
+    const wallet = await recordFunctionMetrics(
+      ServerWallet.create(config ?? payerConfig),
```

Comment:
Is the `await` in the correct place here? Should we await `ServerWallet.create` instead of `recordFunctionMetrics`?

Reply:
Also, does `recordFunctionMetrics` work with promises?

Reply:
> Is the `await` in the correct place here? Should we await `ServerWallet.create` instead of `recordFunctionMetrics`?

I think you're right. I wasn't paying too much attention and while `await recordFunctionMetrics` would still return the `wallet` I don't think metrics were being recorded for the wallet members.

Fixed in 62871e5

Reply:
It looks like `recordFunctionMetrics` uses node `timerify` under the hood. Are we sure that `timerify` can accept promises? If not `recordFunctionMetrics` might not be measuring anything useful here.

Reply:
> It looks like `recordFunctionMetrics` uses node `timerify` under the hood. Are we sure that `timerify` can accept promises? If not `recordFunctionMetrics` might not be measuring anything useful here.

I'm not sure if it handles promises but with the changes in 62871e5 we're now passing in a wallet object instead of a promise so I don't think it should matter?

Reply:
Gotcha! I wonder what we are timing in that case then...

Reply:
> Gotcha! I wonder what we are timing in that case then...

I think the idea is that [it iterates through each function on the wallet and wraps it in a `timerify`](https://github.com/statechannels/statechannels/blob/08add256bf70bffd5551882a6b019938f8670a18/packages/server-wallet/src/metrics.ts#L99). That way we should get metrics on all the `wallet` functions. Not sure if it still works or provides useful information right now though

Reply:
Ah gotcha. That makes sense, thank you for the explanation!

Reply:
This might be an appropriate link here: https://mtlynch.io/code-review-love/#4-answer-questions-with-the-code-itself

(though I appreciate that the resulting conversation is somewhat off topic from the scope of the PR)